polynote-frontend/polynote/cell.js

```diff
@@ -217,6 +204,35 @@ export class CodeCell extends Cell {
         window.addEventListener('resize', this.onWindowResize);
     }
 
+    onChangeModelContent(event) {
+        this.updateEditorHeight();
+        if (this.applyingServerEdits)
+            return;
+
+        // clear the markers on edit
+        // TODO: there might be non-error markers, or might otherwise want to be smarter about clearing markers
+        monaco.editor.setModelMarkers(this.editor.getModel(), this.id, []);
+        const edits = event.changes.flatMap((contentChange) => {
+            if (contentChange.rangeLength && contentChange.text.length) {
+                return [new messages.Delete(contentChange.rangeOffset, contentChange.rangeLength), new messages.Insert(contentChange.rangeOffset, contentChange.text)];
+            } else if (contentChange.rangeLength) {
+              return [new messages.Delete(contentChange.rangeOffset, contentChange.rangeLength)];
+            } else if (contentChange.text.length) {
+                return [new messages.Insert(contentChange.rangeOffset, contentChange.text)];
+            } else return [];
```

Comment:
Hmm, under what circumstances will we get here? If we don't expect it, should we at least log how we got here?

Reply:
I dunno... it's not clear to me whether it would be considered a bug in monaco if it emitted an empty edit. But if it does, I think that's fine; just no reason to do anything with it.